internal/metrics/metrics.go

```diff
@@ -0,0 +1,125 @@
+package metrics
+
+import (
+	"context"
+	"os"
+	"runtime"
+	"time"
+
+	"github.com/shirou/gopsutil/cpu"
+	"github.com/shirou/gopsutil/mem"
+	"github.com/shirou/gopsutil/net"
+	"github.com/shirou/gopsutil/process"
+)
+
+type Metrics struct {
+	ProcessCPU       ProcessCPU
+	CPU              map[string]CPU
+	NIC              map[string]NIC
+	Memory           Memory
+	CPUPercent       float64
+	GarbageCollector GarbageCollector
+}
+
+type GarbageCollector struct {
```

Comment:
(Can be addressed later) Guessing we also want other GC-related metrics here, e.g., `NextGC` and `LastGC`.